File: code/stable/projectile/src/java/Projectile/InputFormat.java
package Projectile;

/** \file InputFormat.java
    \brief Provides the function for reading inputs
*/
import java.util.Arrays;
import java.util.BitSet;
import java.util.Scanner;
import java.io.PrintWriter;
import java.io.FileWriter;
import java.io.File;
import java.util.ArrayList;

public class InputFormat {
    
    /** \brief Reads input from a file with the given file name
        \param filename name of the input file
        \param inParams structure holding the input values
    */
    public static void get_input(String filename, InputParameters inParams) throws Exception {
        Scanner infile;
        infile = new Scanner(new File(filename));
        infile.nextLine();
        inParams.v_launch = Double.parseDouble(infile.nextLine());
        infile.nextLine();
        inParams.theta = Double.parseDouble(infile.nextLine());
        infile.nextLine();
        inParams.p_target = Double.parseDouble(infile.nextLine());
        infile.close();
    }
}

File: examples/projectile/doxygen/projectile_s_l_nol_u_u_v_f/cpp/cpp/InputFormat.cpp
#include "InputFormat.hpp"

#include <algorithm>
#include <fstream>
#include <limits>
#include <string>

using std::ifstream;
using std::string;

void get_input(string filename, float &v_launch, float &theta, float &p_target) {
    ifstream infile;
    infile.open(filename, std::fstream::in);
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> v_launch;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> theta;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> p_target;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile.close();
}
